src/content-scripts/videoConsumption.content.js

```diff
@@ -0,0 +1,248 @@
+            let channelURLElement = document.querySelectorAll("#upload-info.ytd-video-owner-renderer .ytd-channel-name a");
+
+            /* If the element hasn't been populated yet, the query selector will return an empty array */
+            if(channelURLElement.length > 0) {
+                return channelURLElement[0].href;
+            }
+            else {
+                console.log("Video being played is on youtube, but the channel URL couldn't be found");
+                return "";
+            }
+        }
+    }
+
+    /**
+     * Helper function that transmits message with all information from page visit 
+     */
+    function sendVideoDataMessage(playbackDuration, timeStamp) {
+        /* Don't bother sending a message for pages where no video was played. */
+        if(playbackDuration > 0) {
+            let ytChannel = getChannelURL();
+            /* Send message to background page with information about video consumption. */
+            let videoDataMessage = {
+                type: "webScience.videoConsumption.videoData",
+                pageId: pageManager.pageId,
+                url: pageManager.url,
+                referrer: pageManager.referrer,
+                pageVisitStartTime: pageManager.pageVisitStartTime,
+                pageVisitStopTime: timeStamp,
+                playbackDuration,
+                ytChannel,
+                privateWindow: browser.extension.inIncognitoContext
+            };
+            pageManager.sendMessage(videoDataMessage);
+        }
+    }
+
+    /**
+     * Helper function to reset any global variables for the script at the end of a page visit.
+     */
+    function cleanupVariables() {
+        /* Clear the timer in order to stop checking for videos */
+        if(timerId !== 0)
+            clearInterval(timerId);
+        timerId = 0;
+
+        /* Clear the data we have about videos on the page */
+        for (let i = 0; i < videoConsumptions.length; i++) {
+            let vc = videoConsumptions[i];
+            vc.playbackStart = -1;
+            vc.playbackDuration = 0;
+        }
+    }
+
+    let pageVisitStartListener = function ({ timeStamp }) {
+        /* Look for new videos to start listening to, and start the timer ticking to do so continuously */
+        attachVideoListeners();
+        timerId = setInterval(attachVideoListeners, updateInterval);
+    };
+
+    let pageVisitStopListener = function({ timeStamp }) {
+        cleanupPlayingVideos();
+
+        let playbackDuration = aggregatePlaybackDuration();
+
+        sendVideoDataMessage(playbackDuration, timeStamp);
+
+        cleanupVariables();
+    };
+
+    let pageAttentionUpdateListener = function({ timeStamp }) {
+        // console.log("[VC] page attention update");
+    }
+
+    /* Wait for PageManager load */
+    let pageManagerLoaded = function () {
+        pageManager.onPageVisitStart.addListener(pageVisitStartListener);
```

Comment:
If you want to reference `pageManager` rather than `webScience.pageManager`, you'll need something like `let pageManager = null` at the top and `pageManager = webScience.pageManager` here.